src/main/java/org/sonarqube/gradle/AndroidUtils.java

```diff
@@ -147,27 +178,32 @@ private static String getTestBuildType(Project project) {
   }
 
   @Nullable
-  static BaseVariant findVariant(Project project, @Nullable String userConfiguredBuildVariantName) {
+  static AndroidVariantAndExtension findVariantAndExtension(Project project, @Nullable String userConfiguredBuildVariantName) {
     String testBuildType = getTestBuildType(project);
     PluginCollection<AppPlugin> appPlugins = project.getPlugins().withType(AppPlugin.class);
     if (!appPlugins.isEmpty()) {
       AppExtension androidExtension = project.getExtensions().getByType(AppExtension.class);
-      return findVariant(new ArrayList<>(androidExtension.getApplicationVariants()), testBuildType, userConfiguredBuildVariantName);
+      BaseVariant variant = findVariant(new ArrayList<>(androidExtension.getApplicationVariants()), testBuildType, userConfiguredBuildVariantName);
+      return new AndroidVariantAndExtension(androidExtension, variant, userConfiguredBuildVariantName);
```

Comment:
userConfiguredBuildVariantName is not customized in this method, could we remove it from AndroidVariantAndExtension

Reply:
Not sure I understand 🧐 We need it for this logic:
      this.variantProvidedByUser = variant != null && variant.getName().equals(userConfiguredVariantName);
that resides in the constructor of AndroidVariantAndExtension.

Reply:
It is passed in the method @Nullable String userConfiguredBuildVariantName, but returned in the output object.
But with a second look, it seems fine with usage of  the method isVariantProvidedByUser